### DISCUSSION TITLE
Autonomous logger with SoftAP

### DISCUSSION THREAD

**Discussion by steelman** [arendst/Tasmota#17559] [Q&A]
I am looking for instructions how to configure Tasmota to:

1. Wake up periodically to do measurement (this one's well [documented](https://tasmota.github.io/docs/DeepSleep/))
2. Save data on an SD card.
3. Serve them via SoftAP+HTTP upon request (a person comes to a remote location where the sensor is installed, pushes a button to wake it up, connects to the SoftAP and collects the data)

Preferably SoftAP is started only when sensor is woken up manually.

Such setup could collect data in remote locations, where connectivity isn't available for different reasons.

**Comment by barbudor** [comment]
Tasmota is not designed to work with softAP. Only supported for initial configuration. 
For proper operation, tasmota needs to connect to an AP. Access to a NTP server is also mandatory unless one of the supported RTC chip is added

**Reply by steelman** [reply]
> Tasmota is not designed to work with softAP

But it is capable, and I'd like to leverage this capability if possible. And adding an RTC module isn't a problem at all.

**Reply by s-hadinger** [reply]
Not designed means both not capable and not easy to add. Sorry but I don't see an easy path for this

**Reply by steelman** [reply]
Forgive me my insistence/ignorance, if a configuration page can be served, why a data file from an SD card can't? OK, it probably can't be served now, but how big changes need to be? I guess there may be problems with proper authentication, WPA and such. Let's assume I am fine with unencrypted communication.

 Is it possible to serve such data file when connected to a regular AP?

**Reply by barbudor** [reply]
Serving a data file from filesystem is possible
It's soft AP that is not possible without a major restructuration that we won't do
Tasmota is designed for home appliance / automation with a need for a local wifi AP and preferably a mqtt broker

Now you can fork and do whatever you want but you are on your own. Sorry

**Reply by sfromis** [reply]
This is not about whether some concept would be potentially doable within some hardware configuration, but whether it "clicks" with how Tasmota operates. It does not, and thus no straightforward way of adding it.

**Reply by steelman** [reply]
Fair enough.

**Comment by lalo-uy** [comment]
May be you can start the AP from the device that will colect the data. All
phones can do that.
When youpush the buttom tastota starts the wifi and connects.
That scenario coud be investigated.

**Reply by steelman** [reply]
Yes, this is a valid alternative. However, slightly less flexible at least in theory, because if more than one person is to collect data they all have to have their phones configured the same way. OTOH with many loggers to collect data from, phones' configuration wouldn't be cluttered. Definitely worth considering.

BUT, I don't know if smartphones ( Android, not heavily modified) can connect to devices they act as AP for.

**Reply by pkkrusty** [reply]
Discussion #13362 has a hacked driver to allow access to the main Tasmota page even in Soft AP mode. Haven't tested it yet, but might be a next step for you.

As far as I can tell, both iOS and Android can connect to devices they act as AP for. iOS 14.x and Android 9.x verified.

...quick test seems to work on ESP8266. Note that the author used release 12.3.1 as the basis.
![IMG_23D86E7E7A70-1](https://user-images.githubusercontent.com/79770016/220314194-9dd7ba40-37f5-4726-8645-d9734f33912c.jpeg)
